src/DashboardService.Business/Board/GetBoardsCommand.cs

```diff
@@ -3,31 +3,43 @@
 using LT.DigitalOffice.DashboardService.Mappers.Models.Interfaces;
 using LT.DigitalOffice.DashboardService.Models.Dto.Models;
 using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Board.Filters;
-using LT.DigitalOffice.Kernel.Helpers.Interfaces;
 using LT.DigitalOffice.Kernel.Responses;
+using System.Threading;
 using System.Threading.Tasks;
+using LT.DigitalOffice.DashboardService.Models.Db;
+using System.Collections.Generic;
 
 namespace LT.DigitalOffice.DashboardService.Business.Board;
 
 public class GetBoardsCommand : IGetBoardsCommand
 {
   private readonly IBoardRepository _boardRepository;
   private readonly IBoardInfoMapper _boardInfoMapper;
-  private readonly IResponseCreator _responseCreator;
 
   public GetBoardsCommand(
     IBoardRepository repository,
-    IBoardInfoMapper boardInfoMapper,
-    IResponseCreator responseCreator)
+    IBoardInfoMapper boardInfoMapper)
   {
     _boardRepository = repository;
     _boardInfoMapper = boardInfoMapper;
-    _responseCreator = responseCreator;
   }
 
-
-  public Task<FindResultResponse<BoardInfo>> ExecuteAsync(GetBoardsFilter filter)
+  public async Task<FindResultResponse<BoardInfo>> ExecuteAsync(GetBoardsFilter filter, CancellationToken ct)
   {
-    throw new System.NotImplementedException();
+    (List<DbBoard> boards, int totalCount) = await _boardRepository.GetAllAsync(filter, ct);
+
+    if (boards is null || boards.Count == 0)
+    {
+      return new()
+      {
+        Body = new List<BoardInfo>(),
+      };
+    }
```

Comment:
this is unnecessary your method will never return null and .ConvertAll can handle empty lists

Reply:
++